apps/safety/main.cpp

```diff
@@ -11,97 +11,134 @@
 Serial pc(SERIAL_TX, SERIAL_RX, ROVER_DEFAULT_SERIAL_BAUD_RATE);
 
 //CAN initialization 
-CAN can(CAN1_RX, CAN1_TX, ROVER_CANBUS_FREQUENCY); // dont know the actual CAN id's
+CAN can(CAN1_RX, CAN1_TX, ROVER_CANBUS_FREQUENCY); //TODO: dont know the actual CAN id's, find out can id for safety
 CANMsg rxMsg;
-//needs a specific CANID to update values
-constexpr unsigned int EXPECTED_CAN_ID = 0x000;
+//TODO: create possible user commands
+constexpr unsigned int SET_ALERT_LIMIT = 0x00;
+constexpr unsigned int SET_CONFIGURATIONS = 0x01;
 
-//measurement thresholds, needs updating, comparison in celcius now
+//measurement thresholds, needs updating, comparison in celcius 
 constexpr float CURRENT_UPPER_LIM = 1;
 constexpr float CURRENT_LOWER_LIM = 0;
 constexpr float CURVOLTAGE_UPPER_LIM = 1;
 constexpr float VOLTAGE_LOWER_LIM = 0;
 constexpr float TEMP_UPPER_LIM = 1;
-constexpr float TEMP_LOWER_LIM = 0;    
-
-//sample struct for component specific data
-struct elbow_data
-{
-    float shunt_resistance;
-    float max_expected_current;
-    u_int8_t sensor_address; 
+constexpr float TEMP_LOWER_LIM = 0;  
+
+constexpr int NUM_SAFETY_SENSORS = 6;
+enum SENSORNAMES{
+    elbow_sensor,
+    claw_sensor,
+    wrist_sensor,
+    bicep_sensor,
+    allen_key_sensor,
+    turntable_sensor
 };
 
-
 void initCAN();
 //TODO may need more functions to deal with different CAN message contents
 void validateCANMsg(CANMsg *p_newMsg);
 bool inRange(float upper_lim, float lower_lim, float val);
 
 int main()
 {
-    INA_226 elbow_current_sensor(); //example for elbow sensor, may initialize for each component
-    thermistor temp_sensor(TEMP_ANLG_IN); 
+    // SDA Pin, SCL pin, sensor address, shunt resistance, max expected current
+    //TODO: confirm values for sensors
+    ComponentConfig elbow_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL, 0x00, 0.003, 28.75};
+    ComponentConfig claw_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL,0x00, 0.012, 6.325};
+    ComponentConfig wrist_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL,0x00, 0.01, 8.05};
+    ComponentConfig bicep_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL,0x00, 0.015, 5.75};
+    ComponentConfig allen_key_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL,0x00, 0.022, 3.45};
+    ComponentConfig turntable_config = {CUR_SEN_I2C_SDA, CUR_SEN_I2C_SCL,0x00, 0.00075, 60.95};
```

Comment:
Where are all these numbers from? Might be more readable if we have constexprs for them.